manifests/agent/replication.pp

```diff
@@ -72,102 +70,18 @@
   }
 
   if $ensure == 'present' {
-    if $mixin_types {
-      validate_array($mixin_types)
-    }
-
-    if $batch_enabled {
-      validate_bool($batch_enabled)
-    }
-    if $batch_max_wait {
-      validate_integer($batch_max_wait, undef, 1)
-    }
-    if $batch_trigger_size {
-      validate_integer($batch_trigger_size, undef, 1)
-    }
-
-    validate_bool($enabled)
-
-    validate_re($log_level, '^(debug|info|error)$',
-      "${log_level} is not supported for log_level. Allowed values are 'debug', 'info', and 'error'.")
-
-    if $protocol_close_conn {
-      validate_bool($protocol_close_conn)
-    }
-    if $protocol_conn_timeout {
-      validate_integer($protocol_conn_timeout, undef, 1)
-    }
-    if $protocol_http_headers {
-      validate_array($protocol_http_headers)
-    }
-    if $protocol_sock_timeout {
-      validate_integer($protocol_sock_timeout, undef, 1)
-    }
-
-    if $proxy_port {
-      validate_integer($proxy_port, undef, 1)
-    }
-
     if !$resource_type {
```

Comment:
Je kan dit en volgende ifs weglaten als je die gewoon een correct type geeft in de params list en geen default waarde.

Reply:
Updated